File: 2024-2/sesiones-de-clase/sesion-de-clase-20-1/AlumnoIntercambio.cpp
/* 
 * File:   AlumnoIntercambio.cpp
 * Author: erichuiza
 * 
 * Created on November 11, 2024, 3:19 PM
 */

#include "AlumnoIntercambio.h"

AlumnoIntercambio::AlumnoIntercambio() {
    paisOrigen = {};
    universidadOrigen = {};
    duracion = {};
}

AlumnoIntercambio::AlumnoIntercambio(const AlumnoIntercambio& orig) {
    *this = orig;
}

Alumno* AlumnoIntercambio::clonar() {
    return new AlumnoIntercambio(*this);
}

void AlumnoIntercambio::operator=(const AlumnoIntercambio& orig) {
    Alumno::operator =(orig);
    
    paisOrigen = orig.paisOrigen;
    universidadOrigen = orig.universidadOrigen;
    duracion = orig.duracion;
}

void AlumnoIntercambio::cargar(ifstream& archivo) {
    Alumno::cargar(archivo);
    getline(archivo, paisOrigen, ',');
    getline(archivo, universidadOrigen, ',');
    archivo >> duracion;
    archivo.ignore();
}

void AlumnoIntercambio::imprimir(ofstream& archivo) const {
    Alumno::imprimir(archivo);
    archivo << setw(10) << paisOrigen << setw(20) << universidadOrigen << setw(5) << duracion << endl;
}

AlumnoIntercambio::~AlumnoIntercambio() {
}

File: 2024-2/sesiones-de-clase/sesion-de-clase-25/src/AlumnoIntercambio.java

import java.util.Scanner;

/**
 *
 * @author erichuiza
 */
class AlumnoIntercambio extends Alumno {
    private String paisOrigen;
    private String universidadOrigen;
    private int duracion;
    
    public AlumnoIntercambio() {
        super();
    }
    
    public AlumnoIntercambio(char tipo) {
        super(tipo);
    }
    
    public void setPaisOrigen(String paisOrigen) {
        this.paisOrigen = paisOrigen;
    }

    public void setUniversidadOrigen(String universidadOrigen) {
        this.universidadOrigen = universidadOrigen;
    }

    public void setDuracion(int duracion) {
        this.duracion = duracion;
    } 

    public String getPaisOrigen() {
        return paisOrigen;
    }

    public String getUniversidadOrigen() {
        return universidadOrigen;
    }

    public int getDuracion() {
        return duracion;
    }

    @Override
    public void cargar(Scanner scanner) {
        super.cargar(scanner);
        paisOrigen = scanner.next();
        universidadOrigen = scanner.next();
        duracion = scanner.nextInt(); 
    }

    @Override
    public void imprimir() {
        super.imprimir();
        System.out.println("Pais de Origen: " + paisOrigen);
        System.out.println("Universidad: " + universidadOrigen);
        System.out.println("Duración: " + duracion);
    }
}
